Stripe/STPRedirectContext.m

```diff
@@ -196,7 +249,7 @@ - (void)cancel {
 
 - (void)safariViewControllerDidFinish:(__unused SFSafariViewController *)controller {
     stpDispatchToMainThreadIfNecessary(^{
-        [self handleRedirectCompletionWithError:nil
+        [self handleRedirectCompletionWithError:[NSError stp_cardRequiresAdditionalVerificationToProceedError]
```

Comment:
Doesn't this also get called in cases where this error doesn't apply, like if you successfully complete authentication and close the webview?  

AFAIK we can't know whether authentication failed or succeeded here without fetching the PaymentIntent and inspecting its status (this is what `STPPaymentHandler` does)

Reply:
At least in my own testing, all successful and unsuccessful flows ended with a call to `handleURLCallback`, which explicitly triggered the dismissal of the VC. I assumed `safariViewControllerDidFinish:` was implemented only for the use case where the user taps the Done button, mid-flow.

Are you saying we're unable to lean on all web flows to indicate when the VC should be dismissed?

Reply:
Ah, I'm thinking of the case where the user taps the Done button after finishing authentication - my understanding is, at least in testmode, the user ends up here: 
![image](https://user-images.githubusercontent.com/47796191/61982166-b3f82600-afb0-11e9-91c9-f3d5a4b1ab29.png) and they can hit Done if they want instead of the '<- Return to Merchant' button.

If all that's true, I think we just need to continue passing `nil` as the error here (and remove the related `stp_cardRequiresAdditionalVerificationToProceedError` method).

Reply:
Ah, ok. I removed the error and resumed passing `nil` here.